src/portfft/descriptor.hpp

```diff
@@ -452,6 +462,14 @@ class committed_descriptor {
     in_bundle.template set_specialization_constant<detail::SpecConstMultiplyOnLoad>(multiply_on_load);
     in_bundle.template set_specialization_constant<detail::SpecConstMultiplyOnStore>(multiply_on_store);
     in_bundle.template set_specialization_constant<detail::SpecConstApplyScaleFactor>(scale_factor_applied);
+    in_bundle.template set_specialization_constant<detail::SpecConstTakeConjugateOnLoad>(take_conjugate_on_load);
+    in_bundle.template set_specialization_constant<detail::SpecConstTakeConjugateOnStore>(take_conjugate_on_store);
+    if constexpr (std::is_same_v<Scalar, float>) {
+      in_bundle.template set_specialization_constant<detail::SpecConstScaleFactorFloat>(scale_factor);
+    } else {
+      in_bundle.template set_specialization_constant<detail::SpecConstScaleFactorDouble>(scale_factor);
+    }
```

Comment:
Can we encapsulate the choice of spec constant in a function? Something like:
```C++
template<typename Scalar>
constexpr sycl::specialization_id<Scalar> get_spect_constant_scale(){
  if constexpr (std::is_same_v<Scalar, float>) {
    return detail::SpecConstScaleFactorFloat;
  } else{
    return detail::SpecConstScaleFactorDouble;
  }
}
```

Reply:
You might be right, but I would try it. If it does not work you can also try returning a reference.